packages/web-frontend/cypress/e2e/mapOverlays.spec.js

```diff
@@ -3,27 +3,22 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import {
-  constructEveryItemSync,
-  constructIsArrayOf,
-  constructIsEmptyOrSync,
-  constructIsOneOf,
-  hasContent,
-  isBoolean,
-  ObjectValidator,
-} from '@tupaia/utils';
+import { groupBy } from 'lodash';
 
-import config from '../config.json';
+import config from '../__generatedConfig.json';
 import { SNAPSHOT_TYPES } from '../constants';
 import { preserveUserSession } from '../support';
 
-const overlayConfigSchema = {
-  allowEmptyResponse: [constructIsEmptyOrSync(isBoolean)],
-  urls: [hasContent, constructIsArrayOf('string')],
-  snapshotTypes: [
-    hasContent,
-    constructEveryItemSync(constructIsOneOf([SNAPSHOT_TYPES.RESPONSE_BODY])),
-  ],
+const getResponseData = response => {
+  const { measureData, ...data } = response.body;
+
+  if (Array.isArray(measureData)) {
+    // Item order in `measureData` is neither guaranteed nor important.
+    // Group them by org unit for more robust data checking
+    data.e2e_groupedMeasureData = groupBy(measureData, 'organisationUnitCode');
+  }
+
+  return data;
```

Comment:
It would be good to get a second opinion on this... Here we include all response fields in the snapshots. However, many of them are metadata, and some of them are not currently displayed in the front end. So, even if there are any changes in them, users would not notice.

A `measureData` response example:

```json
{
  "chartType": "bar",
  "dashboardGroupId": "88",
  "data": [
    { "name": "ACT", "value": 124 },
    { "name": "NSW", "value": 5729 }
  ],
  "name": "COVID-19 Total Confirmed Cases by State",
  "organisationUnitCode": "AU",
  "period": {
    "earliestAvailable": "20200125",
    "latestAvailable": "20210626",
    "requested": "201701;201702..."
  },
  "showPeriodRange": "all",
  "type": "chart",
  "viewId": "COVID_Total_Cases_By_State"
}
```

Our main focus is in `data`, but we also show other fields in the front end (eg `period.latestAvailable`). So I am not sure if using the whole response is the best way forward:

**Pros**
* No need to worry about updating e2e tests if fields change/are exposed in the front end
* Metadata don't change often, so including them should not frequently result in false negatives (?)
* We currently don't use component (DOM) testing for overlays, so including metadata such as `name` in the snapshots can be kind of a substitute. This does not assert that the metadata are actually rendered correctly, but the same is true for `data` too

**Cons**
* Bigger snapshots
* Tests may fail for unimportant reasons (eg a field that was deprecated)

I was thinking that we can use the "safest" approach for now, and revisit this after we get some feedback from actual usage. Another approach would be adding config that controls this behaviour (eg `includeMetadata`)